nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -310,19 +312,28 @@ public Configuration getConf() {
 
   @VisibleForTesting
   String currentHash() {
-    return client.getRef().getHash();
+    return refreshedClient().getRef().getHash();
   }
 
   @VisibleForTesting
   String currentRefName() {
-    return client.getRef().getName();
+    return refreshedClient().getRef().getName();
   }
 
   @VisibleForTesting
   FileIO fileIO() {
     return fileIO;
   }
 
+  private NessieIcebergClient refreshedClient() {
+    try {
+      client.refresh();
```

Comment:
Alternatively, since the code does not access the `client` field directly now (except for init and refresh), why not call the method simply `client()`?